src/System.Text.Json/tests/WritableJsonApiTests.JsonNumberUnitTests.cs

```diff
@@ -0,0 +1,473 @@
+        {
+            var jsonNumber = new JsonNumber("-1");
+
+            // Getting unsigned types should fail:
+            Assert.False(jsonNumber.TryGetByte(out byte byteResult));
+            Assert.Throws<OverflowException>(() => jsonNumber.GetByte());
+
+            Assert.False(jsonNumber.TryGetUInt16(out ushort ushortResult));
+            Assert.Throws<OverflowException>(() => jsonNumber.GetUInt16());
+
+            Assert.False(jsonNumber.TryGetUInt32(out uint uintResult));
+            Assert.Throws<OverflowException>(() => jsonNumber.GetUInt32());
+
+            Assert.False(jsonNumber.TryGetUInt64(out ulong ulongResult));
+            Assert.Throws<OverflowException>(() => jsonNumber.GetUInt64());
+        }
+
+        [Fact]
+        public static void TestRationalGetMismatches()
+        {
+            var jsonNumber = new JsonNumber("3.14");
+
+            // Getting integer types should fail:
+            Assert.False(jsonNumber.TryGetByte(out byte byteResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetByte());
+
+            Assert.False(jsonNumber.TryGetInt16(out short shortResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetInt16());
+
+            Assert.False(jsonNumber.TryGetInt32(out int intResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetInt32());
+
+            Assert.False(jsonNumber.TryGetInt64(out long longResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetInt64());
+
+            Assert.False(jsonNumber.TryGetSByte(out sbyte sbyteResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetSByte());
+
+            Assert.False(jsonNumber.TryGetUInt16(out ushort ushortResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetUInt16());
+
+            Assert.False(jsonNumber.TryGetUInt32(out uint uintResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetUInt32());
+
+            Assert.False(jsonNumber.TryGetUInt64(out ulong ulongResult));
+            Assert.Throws<FormatException>(() => jsonNumber.GetUInt64());
+        }
+
+        [Theory]
+        [InlineData("0")]
+        [InlineData("-17")]
+        [InlineData("17")]
+        [InlineData("3.14")]
+        [InlineData("-3.1415")]
+        [InlineData("1234567890")]
+        [InlineData("1e400")]
+        [InlineData("1e+100000002")]
+        [InlineData("184467440737095516150.184467440737095516150")]
+        public static void TestToString(string value)
+        {
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetFormattedValue(value);
+            Assert.Equal(value, jsonNumber.ToString());
+        }
+
+        [Fact]
+        public static void TestChangingTypes()
+        {
+            var jsonNumber = new JsonNumber(5);
+            Assert.Equal(5, jsonNumber.GetInt32());
+
+            jsonNumber.SetDouble(3.14);
+            Assert.Equal(3.14, jsonNumber.GetDouble());
```

Comment:
nit: Add Assert.Throws for jsonNumber.GetInt32() after the change of value.

Reply:
I think it is already tested by `TestRationalGetMismatches`.